graphics/macgui/mactext-canvas.cpp

```diff
@@ -80,22 +81,42 @@ void MacTextCanvas::chopChunk(const Common::U32String &str, int *curLinePtr, int
 	int w = getLineWidth(curLine, true);
 	D(9, "** chopChunk before wrap \"%s\"", Common::toPrintable(str.encode()).c_str());
 
+	
 	chunk->getFont()->wordWrapText(str, maxWidth, text, lineContinuations, w);
 
-	if (text.size() == 0) {
+	for (int i = 0; i < (int)text.size(); i++) {
+
+		D(9, "Line Continuations [%d] : %d", i, lineContinuations[i]);
+
+	}
+
+	if (text.empty()) {
 		D(5, "chopChunk: too narrow width, >%d", maxWidth);
-		chunk->text += str;
+
+		if (w < maxWidth) {
+			chunk->text += str;	//Only append if within bounds
+		}
+		
 		getLineCharWidth(curLine, true);
 
 		return;
 	}
 
 	for (int i = 0; i < (int)text.size(); i++) {
+
 		D(9, "** chopChunk result %d \"%s\"", i, toPrintable(text[i].encode()).c_str());
+
 	}
 
 	chunk->text += text[0];
-	_text[curLine].wordContinuation = lineContinuations[0];
+
+	//Ensure line continuations is valid before accesing index 0
+	if (!lineContinuations.empty()) {
+
+		_text[curLine].wordContinuation = lineContinuations[0];
+
+	}
+	
 
```

Comment:
```suggestion
	// Ensure line continuations is valid before accesing index 0
	if (!lineContinuations.empty()) {
		_text[curLine].wordContinuation = lineContinuations[0];
	}
	
```

Reply:
accessing*

Reply:
> accessing*

Indeed! Updated the change above accordingly :)